support-lib/jni/djinni_support.cpp

```diff
@@ -371,6 +371,15 @@ jstring jniStringFromUTF8(JNIEnv * env, const std::string & str) {
     return res;
 }
 
+jstring jniStringFromWString(JNIEnv * env, const std::wstring & str) {
+
+    std::u16string utf16(str.begin(), str.end());;
```

Comment:
I don't object to assuming UTF-16, since we already assume UTF-8 for std::string, which the standard also doesn't say as far as I know.  Assuming 16-bit wchar_t is more questionable.  If it compiles and works, I'd be okay with keeping the assumption (and documenting it) presuming nobody has the need for UCS-4 right now, but UTF-16 in 4-byte chars can work just fine.  If it doesn't work, a clear assert like @j4cbo suggests is a good idea.